src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java

```diff
@@ -154,7 +168,9 @@ public void execute() throws MojoExecutionException {
             getLog().info("About to generate Cucumber report.");
             Reportable report = reportBuilder.generateReports();
 
-            if (checkBuildResult && (report == null || report.getFailedSteps() > 0)) {
+            if (checkBuildResult && (report == null || report.getFailedSteps() > 0 ||
+                    (treatUndefinedAsFailed && report.getUndefinedSteps() > 0) ||
+                    (treatPendingAsFailed && report.getPendingSteps() > 0))) {
```

Comment:
Look for the https://github.com/damianszczepanik/cucumber-reporting project and pull requests there. I don't want to have fragmentation between those two projects, this one is mostly the wrapper for cucumber-reporting so consider creating such change there.

Reply:
> Look for the https://github.com/damianszczepanik/cucumber-reporting project and pull requests there. I don't want to have fragmentation between those two projects, this one is mostly the wrapper for cucumber-reporting so consider creating such change there.

In this particular case, would it make sense to have this logic contained within the reporting project? 
Right now the generated report contains all pass/fail/undefined/pending/etc. results, but in order for Maven to then treat "undefined" or "pending" as failed, this means changing the meaning and intent of "undefined" and "pending" within the reports itself... which doesn't maintain the intent of those statuses from Cucumber's perspective.

Treating the response from the report as a different status within the Maven plugin however does not change the intent of the report results... but does change how Maven treats those responses if the user requested such a change in the POM file.

Since this specific pull request is aimed at changing how Maven behaves with a resultant report, and not how the report itself should be generated, it appears that this change belongs with the maven-cucumber-reporting project, and not with the main cucumber-reporting project.